pkgs/by-name/ni/nixos-rebuild-ng/src/nixos_rebuild/nix.py

```diff
@@ -575,8 +591,21 @@ def switch_to_configuration(
         if not path_to_config.exists():
             raise NRError(f"specialisation not found: {specialisation}")
 
+    r = run_wrapper(
+        ["test", "-d", "/run/systemd/system"],
+        remote=target_host,
+        check=False,
+    )
+    cmd = []
+    if not r.returncode:
+        logger.debug(
+            "skipping systemd-run to switch configuration since systemd is "
+            + "not working in target host"
+        )
+        cmd = SWITCH_TO_CONFIGURATION_CMD_PREFIX
```

Comment:
You are logging that we are skipping systemd-run, but then you still prepend SWITCH_TO_CONFIGURATION_CMD_PREFIX to the command here?

Reply:
Nice catch. Thankfully the logic was correct (it would be really painfull to fix all tests again), it is just the debug log that was wrong.